agent/engine/serviceconnect/version_linux.go

```diff
@@ -0,0 +1,77 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package serviceconnect
+
+import (
+	"regexp"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/aws/amazon-ecs-agent/agent/logger"
+	"github.com/aws/amazon-ecs-agent/agent/logger/field"
+)
+
+const (
+	appnetInterfaceV1          = "v1"
+	serviceConnectCapabilityV1 = "ecs.service-connect.v1"
+
+	appNetInterfaceVersionPattern = "^v(\\d*)$"
+)
+
+var (
+	supportedAppnetInterfaceVerToCapability = map[string][]string{
+		appnetInterfaceV1: {
+			serviceConnectCapabilityV1,
+		},
+	}
+)
+
+// getSupportedAppnetInterfaceVersions returns the all the supported AppNet interface versions
+// by ECS Agent from highest to lowest version
+func getSupportedAppnetInterfaceVersions() []string {
+	supportedAppnetInterfaceVersions := make([]string, 0, len(supportedAppnetInterfaceVerToCapability))
+	for supportedAppnetInterfaceVersion := range supportedAppnetInterfaceVerToCapability {
+		allowedVersionPattern, err := regexp.Compile(appNetInterfaceVersionPattern)
+		if err != nil {
+			logger.Warn("Unable to compile regex pattern:", logger.Fields{
+				field.Error: err,
+			})
+		}
+		if !allowedVersionPattern.MatchString(supportedAppnetInterfaceVersion) {
+			logger.Warn("Appnet interface version does not match the expected pattern, ex: `v1`")
+		}
+		supportedAppnetInterfaceVersions = append(supportedAppnetInterfaceVersions, supportedAppnetInterfaceVersion)
+	}
+	sort.Slice(
+		supportedAppnetInterfaceVersions,
+		func(i, j int) bool {
+			return sortVersion(supportedAppnetInterfaceVersions[i]) > sortVersion(supportedAppnetInterfaceVersions[j])
+		},
+	)
+	return supportedAppnetInterfaceVersions
+}
+
+// sortVersion returns a version sort key in numeric order.
+// For example, v20, v11, v10, v2, v1,...
+func sortVersion(version string) int {
```

Comment:
I'd probably rename it - this function itself simply parses the int value from version string, and doesn't do any sorting on its own. Maybe `getVersionNumber` or something more appropriate.